include/Reader/readerir.h

```diff
@@ -1484,16 +1501,21 @@ class GenIR : public ReaderBase {
   /// \returns    LLVM type that models the built-in object type.
   llvm::Type *getBuiltInObjectType();
 
-  /// Get the LLVM type for an array of references.
+  /// Get the LLVM type for an array of given element type.
   ///
   /// Used when we know that some type must be an array but our local
   /// type information thinks otherwise.
   ///
-  /// \returns    LLVM type that models array of references.
-  llvm::PointerType *getArrayOfReferenceType();
+  /// \param ElementTy                 The desired element type for the array, 
+  ///                                  or if null, use array of references.
+  /// \returns                         LLVM type that models the desired array
+  llvm::PointerType *getArrayOfElementType(llvm::Type *ElementTy = nullptr);
```

Comment:
Agreed for getArrayOfElementType. However in the case of isArrayType and ensureIsArray, 
we want to allow the user to specify a "don't care" for the element type. 
I've adjusted the code accordingly.